modules/evolution_techBidAdapter.js

```diff
@@ -0,0 +1,113 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+
+const BIDDER_CODE = 'e_volution';
+const AD_URL = 'https://service.e-volution.ai/?c=o&m=multi';
+const URL_SYNC = 'https://service.e-volution.ai/?c=o&m=sync';
+const NO_SYNC = true;
+
+function isBidResponseValid(bid) {
+  if (!bid.requestId || !bid.cpm || !bid.creativeId ||
+    !bid.ttl || !bid.currency) {
+    return false;
+  }
+  switch (bid['mediaType']) {
+    case BANNER:
+      return Boolean(bid.width && bid.height && bid.ad);
+    case VIDEO:
+      return Boolean(bid.vastUrl);
+    case NATIVE:
+      return Boolean(bid.native && bid.native.title && bid.native.image && bid.native.impressionTrackers);
+    default:
+      return false;
+  }
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+  noSync: NO_SYNC,
+
+  isBidRequestValid: (bid) => {
+    return Boolean(bid.bidId && bid.params && !isNaN(bid.params.placementId));
+  },
+
+  buildRequests: (validBidRequests = [], bidderRequest) => {
+    let winTop = window;
+    let location;
+    try {
+      location = new URL(bidderRequest.refererInfo.referer)
+      winTop = window.top;
+    } catch (e) {
+      location = winTop.location;
+      utils.logMessage(e);
+    };
+    let placements = [];
+    let request = {
+      'deviceWidth': winTop.screen.width,
+      'deviceHeight': winTop.screen.height,
+      'language': (navigator && navigator.language) ? navigator.language : '',
```

Comment:
Replace `navigator.language` with `navigator.language.split('-')[0]`. Line 56 can then be removed.

If there's no '-', then `split('-')[0]` will be the whole string.

Reply:
+

Reply:
Sorry, I didn't mean that you should remove the ternary operator. If `navigator.language` is null, calling `split` will throw an error. Replace with this:
```
 (navigator && navigator.language) ? navigator.language.split('-')[0] : ''
```

Reply:
My fault, had to think about it. I've made this change

Reply:
The condition you were using originally was safer. This latest one assumes that `navigator` will always have a `language` property. But should be fine either way

Reply:
+